cmd/influx_inspect/export/export.go

```diff
@@ -209,42 +275,27 @@ func (cmd *Command) write() error {
 		w = gzw
 	}
 
-	s, e := time.Unix(0, cmd.startTime).Format(time.RFC3339), time.Unix(0, cmd.endTime).Format(time.RFC3339)
-	fmt.Fprintf(w, "# INFLUXDB EXPORT: %s - %s\n", s, e)
-
-	// Write out all the DDL
-	fmt.Fprintln(w, "# DDL")
-	for key := range cmd.manifest {
-		keys := strings.Split(key, string(os.PathSeparator))
-		db, rp := influxql.QuoteIdent(keys[0]), influxql.QuoteIdent(keys[1])
-		fmt.Fprintf(w, "CREATE DATABASE %s WITH NAME %s\n", db, rp)
-	}
-
-	fmt.Fprintln(w, "# DML")
-	for key := range cmd.manifest {
-		keys := strings.Split(key, string(os.PathSeparator))
-		fmt.Fprintf(w, "# CONTEXT-DATABASE:%s\n", keys[0])
-		fmt.Fprintf(w, "# CONTEXT-RETENTION-POLICY:%s\n", keys[1])
-		if files, ok := cmd.tsmFiles[key]; ok {
-			fmt.Fprintf(cmd.Stdout, "writing out tsm file data for %s...", key)
-			if err := cmd.writeTsmFiles(w, files); err != nil {
-				return err
-			}
-			fmt.Fprintln(cmd.Stdout, "complete.")
-		}
-		if _, ok := cmd.walFiles[key]; ok {
-			fmt.Fprintf(cmd.Stdout, "writing out wal file data for %s...", key)
-			if err := cmd.writeWALFiles(w, cmd.walFiles[key], key); err != nil {
-				return err
-			}
-			fmt.Fprintln(cmd.Stdout, "complete.")
+	// mw is our "meta writer" -- the io.Writer to which meta/out-of-band data
+	// like comments will be sent.  If the lponly flag is set, we mw will be
+	// ioutil.Discard which effectively filters out comments and any other
+	// non-line protocol data.
+	//
+	// Otherwise, mw is set to the same writer as the actual DDL and line
+	// protocol DML which will cause the comments to be intermixed with the
+	// data..
+	//
+	mw := func() io.Writer {
+		if cmd.lponly {
+			return ioutil.Discard
 		}
-	}
-	return nil
+		return w
+	}()
```

Comment:
Saves a couple lines of code and simpler...
```
mw := w
if cmd.lponly {
   mw = ioutil.Discard
}
```

Reply:
Done.  Just so you know, I did think about this.  My intention here is that I feel like with:

```
mw := w
if cmd.lponly {
  mw = ioutil.Discard
}
```
as you read it, you have to track the value of `mw` through the condition and the variable name is intermixed with the logic for assigning itself.   Personally, I like assigning `mw` only once and keeping all of the logic that dictates is value separate; it feels easier to read to me.  So I think about it -- if maybe came to a different conclusion.